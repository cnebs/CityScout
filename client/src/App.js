import React, { Component } from 'react';
import fetch from 'node-fetch';
import CardContainer from './components/CardContainer.js'
import Navbar from './components/Navbar'
// import './App.css';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      eventsPlaceHolder: [
        {
            source_API: 'TicketMaster',
            name: 'Hayes Carll',
            url: 'http://www.ticketsnow.com/InventoryBrowse/TicketList.aspx?PID=2718472',
            event_id: 'Z7r9jZ1Aejbow',
            time_start: '2019-08-11T02:00:00Z',
            time_end: null,
            category: 'Music',
            image: 'https://s1.ticketm.net/dam/a/fc1/e7affb5a-4ba1-4e6f-8aad-29c79f4a6fc1_68981_RECOMENDATION_16_9.jpg',
            venue: 'Gruene Hall',
            location: 'New Braunfels',
            price_min: null,
            price_max: null,
            description: null
          },
          {
            source_API: 'TicketMaster',
            name: 'Trevor Cannon',
            url: 'http://www.ticketsnow.com/InventoryBrowse/TicketList.aspx?PID=2718472',
            event_id: 'Z7r9jZ1Aejboz',
            time_start: '2019-08-11T02:00:00Z',
            time_end: null,
            category: 'Music',
            image: 'https://s1.ticketm.net/dam/a/fc1/e7affb5a-4ba1-4e6f-8aad-29c79f4a6fc1_68981_RECOMENDATION_16_9.jpg',
            venue: 'Gruene Hall',
            location: 'New Braunfels',
            price_min: null,
            price_max: null,
            description: null
          },
          {
            source_API: 'TicketMaster',
            name: 'Hadley Crowl',
            url: 'http://www.ticketsnow.com/InventoryBrowse/TicketList.aspx?PID=2718472',
            event_id: 'Z7r9jZ1Aejbot',
            time_start: '2019-08-11T02:00:00Z',
            time_end: null,
            category: 'Music',
            image: 'https://s1.ticketm.net/dam/a/fc1/e7affb5a-4ba1-4e6f-8aad-29c79f4a6fc1_68981_RECOMENDATION_16_9.jpg',
            venue: 'Gruene Hall',
            location: 'New Braunfels',
            price_min: null,
            price_max: null,
            description: null
          }
    ]
    };
    this.api = `http://localhost:8000/api/example`;
  }
  componentDidMount() {

    //This will be a post request, that expects location and returns cardData
    // fetch(this.api)
    //   .then(res => res.json())
    //   .then(events => {
    //     this.setState({ eventsPlaceHolder: events.data });
    //   });

  }

  render() {
    return (
      <>

        <h1>Welcome to Blue Ocean!</h1>
        <Navbar />
        <CardContainer events={this.state.eventsPlaceHolder}/>
      </>
    );
  }
}
