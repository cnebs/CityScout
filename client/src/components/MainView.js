import React from 'react';
import CardContainer from './CardContainer.js';
import moment from 'moment';
import spinner from '../../public/spinner.gif';
import { Grow } from '@material-ui/core/';

// import fetch from 'node-fetch';

// import { makeStyles } from '@material-ui/core/styles';
import { Paper, Typography, Grid, Container } from '@material-ui/core/';

// const useStyles = makeStyles(theme => ({
// }));

// const today = moment().format('dddd')
// const tomorrow = moment().add(1, 'days').format('dddd')
const tomorrowPlusPlus = moment()
  .add(2, 'days')
  .format('dddd');

export default function MainView({
  name,
  loaded,
  events,
  eventsToday,
  eventsTomorrow,
  eventsTomorrowPlusPlus,
  handleCardActionClick
}) {
  // const classes = useStyles();

  return (
    <>
      {loaded ? (
        <div>
          <Container maxWidth="lg" align="center">
            <Grow in={true} timeout={400}>
              <Typography
                variant="h3"
                color="textSecondary"
                style={{ marginTop: 125, marginBottom: 50 }}
              >
                {name
                  ? `Hello ${name.first_name}, here are your top picks`
                  : `Top Picks For You`}
              </Typography>
            </Grow>
            <Grid container>
              <CardContainer
                event={events[0]}
                day={'Today'}
                animationTime={400}
                handleCardActionClick={handleCardActionClick}
              />
              <CardContainer
                event={events[1]}
                day={'Tomorrow'}
                animationTime={600}
                handleCardActionClick={handleCardActionClick}
              />
              <CardContainer
                event={events[2]}
                day={tomorrowPlusPlus}
                animationTime={800}
                handleCardActionClick={handleCardActionClick}
              />
            </Grid>
            <Button
              variant="outlined"
              className={classes.margin}
              onClick={handleClick}
            >
              Open success snackbar
            </Button>
            <Snackbar
              anchorOrigin={{
                vertical: 'bottom',
                horizontal: 'left'
              }}
              open={open}
              autoHideDuration={4000}
              onClose={handleClose}
            >
              <SnackbarContent
                onClose={handleClose}
                variant="success"
                message="This is a success message!"
              />
            </Snackbar>
          </Container>
        </div>
      ) : (
        <div
          style={{
            paddingTop: '20vh',
            display: 'flex',
            justifyContent: 'center'
          }}
        >
          <img src={spinner} alt="loading events..." />
        </div>
      )}
    </>
  );
}

// import React from 'react';
// import PropTypes from 'prop-types';
// import clsx from 'clsx';
// import Button from '@material-ui/core/Button';
// import CheckCircleIcon from '@material-ui/icons/CheckCircle';
// import CloseIcon from '@material-ui/icons/Close';
// import { amber, green } from '@material-ui/core/colors';
// import IconButton from '@material-ui/core/IconButton';
// import Snackbar from '@material-ui/core/Snackbar';
// import SnackbarContent from '@material-ui/core/SnackbarContent';
// import WarningIcon from '@material-ui/icons/Warning';
// import { makeStyles } from '@material-ui/core/styles';

// const variantIcon = {
//   success: CheckCircleIcon
// };

// const useStyles1 = makeStyles(theme => ({
//   success: {
//     backgroundColor: green[600],
//   },
//   icon: {
//     fontSize: 20,
//   },
//   iconVariant: {
//     opacity: 0.9,
//     marginRight: theme.spacing(1),
//   },
//   message: {
//     display: 'flex',
//     alignItems: 'center',
//   },
// }));

// function MySnackbarContentWrapper(props) {
//   const classes = useStyles1();
//   const { className, message, onClose, variant, ...other } = props;
//   const Icon = variantIcon[variant];

//   return (
//     <SnackbarContent
//       className={clsx(classes[variant], className)}
//       aria-describedby="client-snackbar"
//       message={
//         <span id="client-snackbar" className={classes.message}>
//           <Icon className={clsx(classes.icon, classes.iconVariant)} />
//           {message}
//         </span>
//       }
//       action={[
//         <IconButton key="close" aria-label="close" color="inherit" onClick={onClose}>
//           <CloseIcon className={classes.icon} />
//         </IconButton>,
//       ]}
//       {...other}
//     />
//   );
// }

// MySnackbarContentWrapper.propTypes = {
//   className: PropTypes.string,
//   message: PropTypes.string,
//   onClose: PropTypes.func,
//   variant: PropTypes.oneOf(['error', 'info', 'success', 'warning']).isRequired,
// };

// export default function CustomizedSnackbars() {

//   return (
//     <div>

//     </div>
//   );
// }
