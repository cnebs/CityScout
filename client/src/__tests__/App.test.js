import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import App from '../App';
import React from 'react';
import NavBar from '../components/Navbar';
import MainView from '../components/MainView';

describe('Top level App component', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<App />);
  });
  test('it renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('it renders a single NavBar child', () => {
    expect(wrapper.find(NavBar)).toHaveLength(1);
  });
  test('it renders a single MainView child when / route is hit', () => {
    let mountedWrapper = mount(
      <MemoryRouter initialEntries={['/']}>
        <App />
      </MemoryRouter>
    );
    expect(mountedWrapper.find(MainView)).toHaveLength(1);
  });
  test('it has the correct state shape', () => {
    expect(wrapper.state().eventsAll).toBeInstanceOf(Object);
  });
});
