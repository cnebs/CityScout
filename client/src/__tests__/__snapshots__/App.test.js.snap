// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top level App component it renders without crashing 1`] = `ShallowWrapper {}`;
